src/sources/socket/mod.rs

```diff
@@ -142,44 +156,40 @@ impl SourceConfig for SocketConfig {
                 )
             }
             Mode::Udp(config) => {
-                let host_key = config
-                    .host_key()
-                    .clone()
-                    .unwrap_or_else(|| log_schema().host_key().to_string());
```

Comment:
I think we still need this logic somewhere, to account for the fallback to global behavior. [Ex: the file source](https://github.com/vectordotdev/vector/pull/15174/files#diff-f0af3092b3491929b69c2ae1603b4a9aee2d0981e6b00e9a2d47de3d59c483b8R420-R428)

Reply:
This logic is moved into the function that's called from here: https://github.com/vectordotdev/vector/blob/dh%2Flog_namespaces_for_socket_sources/src/sources/socket/udp.rs#L227